jina/drivers/search.py

```diff
@@ -61,24 +61,25 @@ def __init__(self, is_update: bool = True, traversal_paths: Tuple[str] = ('m'),
         super().__init__(traversal_paths=traversal_paths, *args, **kwargs)
         self._is_update = is_update
 
-    def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        miss_idx = []  #: missed hit results, some search may not end with results. especially in shards
-        for idx, retrieved_doc in enumerate(docs):
-            serialized_doc = self.exec_fn(retrieved_doc.id)
-            if serialized_doc:
-                r = Document(serialized_doc)
-                if self._is_update:
-                    retrieved_doc.update(r)
+    def _apply_all(self, doc_sequences: Iterable['DocumentSet'], *args, **kwargs) -> None:
```

Comment:
It would be nice to have less nested loops and if-statements.

Reply:
True. This function will be refactored in a next step anyhow, where we enable batching on the Executor level.